server/webapp/WEB-INF/rails.new/app/controllers/admin/templates_controller.rb

```diff
@@ -134,10 +136,12 @@ def load_templates from
   end
 
   def load_cruise_config
-    result = HttpLocalizedOperationResult.new
-    assert_load :cruise_config, go_config_service.loadCruiseConfigForEdit(current_user, result)
-    unless result.isSuccessful()
-      render_localized_operation_result result
+    if security_service.isUserAdmin(current_user) || security_service.isAuthorizedToViewAndEditTemplates(current_user)
```

Comment:
Well, for index, we need to load the cruise_config for an admin or template admin, and not for template view users. But we need to allow the template view users to reach the action index. If we don't make this check, then it throws an error for template view user in the template listing page.